### DISCUSSION TITLE
Ability to modify ratings on the fly

### DISCUSSION THREAD

**Discussion by Z-Diezel** [projectM-visualizer/projectm#721] [Q&A] [answered]
I can't see a way to modify the rating of a preset without opening the milk file in notepad. I remember using the '+' and '-' keys in Winamp while the preset was active and it would edit the rating on the fly. There was also the ability to display text overlays showing the preset name and rating. Is there any way to do this yet?

**Comment by kblaschke** [comment, marked as answer]
Ratings support was removed in version 4.0 as the feature wasn't even properly implemented in Milkdrop. To elaborate on this, we talked about a few ways how ratings might be useful, but we couldn't find any proper way to make it work for all different use cases, especially with large preset collections. One major issue is that the rating is saved in the .milk file itself, which might not be writeable for the current user, e.g. on Linux or when loaded from a network or read-only storage. So the ratings would have to saved elsewhere, making them non-portable and adding too much infrastructure needs to the library.

In the end, we decided to remove ratings and add a proper playlist manager into the new SDL app. It's not finished yet, but we're working on it. This will enable you to create playlists with your favorite presets, and even multiple ones to match different music styles. We think it's way easier to understand and use than the obscure 1 to 5 rating system per preset. You couldn't even filter presets by rating, and the random selection was very fuzzy with large collections, many users complaining about low rated presets still appearing and high rated ones still not often enough. With playlists, every user can just define how often a preset should be shown by how often it appears in the playlist and don't add unwanted presets in the first place.
